product_uom_updatable/models/product_template.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2020 Acsone SA/NV
+# Copyright 2020 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from itertools import groupby
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class ProductTemplate(models.Model):
+
+    _inherit = "product.template"
+
+    def write(self, vals):
+        uom_id = vals.pop("uom_id", False)
+        uom_po_id = vals.pop("uom_po_id", False)
+        if uom_id:
+            self._update_uom(uom_id, "uom_id")
+        if uom_po_id:
+            self._update_uom(uom_po_id, "uom_po_id")
+        res = super().write(vals)
+        return res
+
+    def _update_uom(self, uom_id, field_name):
+        uom_obj = self.env["uom.uom"]
+        sorted_items = sorted(self, key=lambda r: r[field_name])
+        for key, products_group in groupby(sorted_items, key=lambda r: r[field_name]):
+            product_list = list(products_group)
+            product_ids = [product.id for product in product_list]
+            new_uom = uom_obj.browse(uom_id)
+            if (
+                key.category_id == new_uom.category_id
+                and key.factor_inv == new_uom.factor_inv
+            ):
+                self.env.cr.execute(
```

Comment:
TBH I don't get why you pop the value and the write it via SQL instead of leaving it in `values`. :thinking: 
Also, couldn't you do this check in a constraint?

Reply:
before writing anything we want to check which parameter (uom_id or uom_po_id) we have in vals so we know what we want to update. We can leave it in values and pass it to this method, but still we will have to check what's inside if we want to use it in query.